core/src/test/java/hudson/model/JobTest.java

```diff
@@ -26,24 +27,21 @@
     @Test
     public void testSetDisplayName() throws Exception {
        final String displayName = "testSetDisplayName";
-
-       StubJob j = new StubJob();      
+       Job<?,?> j = Mockito.mock(Job.class, Mockito.CALLS_REAL_METHODS);
+       Mockito.doNothing().when(j).save();
        // call setDisplayNameFromRequest
        j.setDisplayNameOrNull(displayName);
        
        // make sure the displayname has been set
        assertEquals(displayName, j.getDisplayName());
     }
-
+    
     @Test
-    public void testSetDisplayNameZeroLength() throws Exception {
```

Comment:
It looks like this test was removed and a different test was added. Might make sense to document reason for the same

Reply:
@StefanSpieker could you address this bit?